packages/ui/components/calendar/components/event/EventList.tsx

```diff
@@ -0,0 +1,108 @@
+import shallow from "zustand/shallow";
+
+import dayjs from "@calcom/dayjs";
+
+import { useCalendarStore } from "../../state/store";
+import { Event } from "./Event";
+
+type Props = {
+  day: dayjs.Dayjs;
+};
+
+export function EventList({ day }: Props) {
+  const { startHour, events, eventOnClick } = useCalendarStore(
+    (state) => ({
+      startHour: state.startHour,
+      events: state.events,
+      eventOnClick: state.onEventClick,
+    }),
+    shallow
+  );
+
+  return (
+    <>
+      {events
+        .filter((event) => {
+          return dayjs(event.start).isSame(day, "day") && !event.allDay; // Filter all events that are not allDay and that are on the current day
+        })
+        .map((event, idx, eventsArray) => {
+          let width = 90;
```

Comment:
I can imagine that if we'd add these calculations inside a child component (meaning we turn the event wrapper div into a component as well, could be in this file), that we actually improve the performance because React will then only recalculate one child if that changes. 

Perhaps down the line we can even memoize this more, but let's see how that performs first. What do you think?

Reply:
I agree I will create an issue once this a merged so we can do this in a followup